cube_helper/fix_known.py

```diff
@@ -0,0 +1,180 @@
+        self.cube_attributes_required = {}
+
+    def is_fix_needed(self):
+        """
+        Check if this fix applies to the object's cube by checking that all of
+        the key-value pairs in `self.cube_attributes_required` occur in
+        `cube.attributes`.
+
+        Returns:
+            True if this fix should be applied to the cube that it was
+            instantiated with.
+        """
+        for key, value in six.iteritems(self.cube_attributes_required):
+            if key in self.cube.attributes:
+                if self.cube.attributes[key] != value:
+                    return False
+            else:
+                return False
+        # Haven't found anything that doesn't match
+        return True
+
+
+class FixCmip6CasFgoals(FixKnownIssueIdentifyAttributes):
+    """
+    Fix the latitude and longitude bounds in mip_era: CMIP6 institution_id: CAS
+    source_id: FGOALS-f3-L, which are not contiguous and use
+    `iris.coords.Coord.guess_bounds()` to calculate new bounds for these two
+    coordinates.
+
+    Args:
+        cube: the cube to check, and if necessary fix.
+    """
+    def __init__(self, cube):
+        super(FixCmip6CasFgoals, self).__init__(cube)
+        self.cube_attributes_required = {
+            'mip_era': 'CMIP6',
+            'institution_id': 'CAS',
+            'source_id': 'FGOALS-f3-L'
+        }
+
+    def fix_cube(self):
+        """
+        Delete the existing faulty bounds on the latitude and longitude
+        coordinates and then calculate new bounds.
+        """
+        super(FixCmip6CasFgoals, self).fix_cube()
+        self.cube.coord('latitude').bounds = None
+        self.cube.coord('latitude').guess_bounds()
+        self.cube.coord('longitude').bounds = None
+        self.cube.coord('longitude').guess_bounds()
+
+
+class FixCmip6FioqlnmFioesm20Historical(FixKnownIssueIdentifyAttributes):
+    """
+    Fix the latitude and longitude bounds in mip_era: CMIP6 institution_id:
+    FIO-QLNM source_id: FIO-ESM-2-0 experiment: historical, which are different
+    to other FIO-ESM-2-0 experiments and are not monotonic, preventing latitude
+    from being a dimension coordinate. `iris.coords.Coord.guess_bounds()`
+    calculates bounds for these two coordinates that are identical to the
+    bounds in other experiments and allow latitude to be promoted to a
+    dimension coordinate.
+
+    Args:
+        cube: the cube to check, and if necessary fix.
+    """
+    def __init__(self, cube):
+        super(FixCmip6FioqlnmFioesm20Historical, self).__init__(cube)
+        self.cube_attributes_required = {
+            'mip_era': 'CMIP6',
+            'institution_id': 'FIO-QLNM',
+            'source_id': 'FIO-ESM-2-0',
+            'experiment_id': 'historical'
+        }
+
+    def fix_cube(self):
+        """
+        Delete the existing faulty bounds on the latitude and longitude
+        coordinates and then calculate new bounds.
```

Comment:
... and promote the latitude coordinate to a dimension coordinate?

Reply:
Ooops, well spotted! Copy and paste error. 3264b34 improves the docstring.